### DISCUSSION TITLE
Compact Recipe View

### DISCUSSION THREAD

**Discussion by quisquous** [mealie-recipes/mealie#3382] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Mealie's recipe instructions view is incredibly tall on both desktop and mobile. Each "step box" is 118px tall for 17px of text. For example, if a recipe has ~8 steps, I can't fit it on my maximized monitor, and mobile can only see about ~4 steps. This is largely due to the padding and the large (unnecessary to me) "Step: N" text. 

I would like to be able to see more of the recipe on screen at once and not have to scroll. Alternatively, I could just put all of my recipe instructions into "Step: 1" but this seems less good.

current: 
![image](https://github.com/mealie-recipes/mealie/assets/462317/ce8ddb1e-a860-4f99-bcff-ca3ad5ed5dff)

desired:
![image](https://github.com/mealie-recipes/mealie/assets/462317/cb99ed7d-9ad4-4494-9d82-5e4209631ef3)

### Please provide a concise description of the feature that would resolve your issue.

Possible ideas:

* A "compact view" option to remove "Step: 1" text and reduce the padding.
* Custom CSS would also work here: https://github.com/mealie-recipes/mealie/discussions/1722
* Hide the "Step: 1" text entirely when there is only one step for a section, and then I can do my own formatting in markdown

### Please consider and list out some caveats or tradeoffs made in your design decision

Any of these things come with extra maintenance burdens.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by brcarls** [comment]
I would also appreciate a more compact view.    My mouse is a mess from having to scroll while I'm cooking :)

**Comment by bartkummel** [comment]
I totally agree. The "Step N" header is useless and takes up precious screen space. On top of this, I'd like to see the behavior changed when I click/tap a specific step:

- Currently, when you click/tap a step, it is marked "done", and only the "Step N" header stays visible and the actual instructions are hidden.
- In practice, I regularly do things in a different order than the recipe. I _do_ like to be able to mark the step I'm currently at, but I don't like it that instructions are hidden.

Proposal: instead of marking a step as done when tapping/clicking it, highlight it as "current". This can be done with a subtle border or background color, or adding some indicator in front of it. Also, consider assigning keyboard shortcuts to move to the next or previous step.

**Comment by eneiss** [comment]
I also agree, it would make sense to me to get rid of the "Step X" header when there's no title to the current step.
On top of being more compact and efficient, I believe it would make the actual headers and the overall structure easier to see.
